components/OfferList.tsx

```diff
@@ -0,0 +1,105 @@
+import Link from "next/link";
+import React, { Fragment } from "react";
+import { Offer } from "../lib/model/Offer";
+import { OfferType, QuestionDefinition } from "../lib/shared";
+
+const OfferParameter: React.FC<{
+	parameter: Offer["parameters"][0];
+	question: QuestionDefinition;
+}> = ({ parameter, question }) => {
+	return (
+		<div key={parameter.id} className="flex flex-col mt-2">
+			<p className="text-sm font-bold">{question.question}</p>
+			<p className="text-sm">
+				{question.type === "district" || question.type === "checkbox" ? (
+					<>
+						{parameter.values.map((value, i) => {
+							const isLast = i === parameter.values.length - 1;
+							const requiresSpecification =
+								question.type === "checkbox" &&
+								(question.options.find((it) => it.value === value.value)
+									?.requireSpecification ??
+									false);
+							return (
+								<Fragment key={value.id}>
+									<span>
+										{value.value}
+										{requiresSpecification && ` (${value.specification})`}
+									</span>
+									{!isLast && ", "}
+								</Fragment>
+							);
+						})}
+					</>
+				) : question.type === "radio" ? (
+					<>
+						{parameter.value}
+						{(question.options.find((it) => it.value === parameter.value)
+							?.requireSpecification ??
+							false) &&
+							` (${parameter.specification})`}
+					</>
+				) : question.type === "date" ? (
+					<>
+						{parameter.value} {/* TODO */}
+					</>
+				) : (
+					<>{parameter.value}</>
+				)}
+			</p>
+		</div>
+	);
+};
+
+const OfferList: React.FC<{
```

Comment:
separated file? 1 component per file?

Reply:
I kept the OfferParameter here, because its probably not going to be used anywhere else. We can always move it out